### Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [5]:
data= {
    "student_id":[1,2,3,4,5,6,7,8,9,10],
    "time_of_study":[5,6,7,8,9,10,11,12,13,14],
    "final_score":[4.5,5.6,6.6,7.2,8.1,8.5,8.6,9.1,9.3,9.5]
}

In [7]:
import pandas as pd
df=pd.DataFrame(data)
df

student_id  time_of_study  final_score
0           1              5          4.5
1           2              6          5.6
2           3              7          6.6
3           4              8          7.2
4           5              9          8.1
5           6             10          8.5
6           7             11          8.6
7           8             12          9.1
8           9             13          9.3
9          10             14          9.5

In [10]:
## Calculating Pearson Correlation
df[["time_of_study","final_score"]].corr("pearson")

time_of_study  final_score
time_of_study       1.000000     0.959679
final_score         0.959679     1.000000

##  Interpretation
Based on the `Pearson` correlation coefficient values you provided, there is a very strong positive correlation `(r = 0.959679`) between the amount of time students spend studying for an exam and their final exam scores. This indicates that as the time spent studying increases, the final exam score tends to increase as well. The value of `r = 0.959679`indicates a very strong correlation, which means that this relationship is likely to be important and reliable.

## Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [12]:
data = {
    "amount_of_sleep":[4,5,5,5.2,5.5,5.8,5.1,4,6,6.1,7,4.5,4.6],
    "jobs_statisfaction_level":[2,3,3,4,5,6,4,7,8,9,10,2,3]
}

In [13]:
pd.DataFrame(data)

amount_of_sleep  jobs_statisfaction_level
0               4.0                         2
1               5.0                         3
2               5.0                         3
3               5.2                         4
4               5.5                         5
5               5.8                         6
6               5.1                         4
7               4.0                         7
8               6.0                         8
9               6.1                         9
10              7.0                        10
11              4.5                         2
12              4.6                         3

In [15]:
pd.DataFrame(data).corr(method="spearman")

amount_of_sleep  jobs_statisfaction_level
amount_of_sleep                  1.000000                  0.768067
jobs_statisfaction_level         0.768067                  1.000000

## Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [7]:
import numpy as np
from scipy import stats

# Generate random data for exercise hours and BMI
np.random.seed(123) # For reproducibility
exercise_hours = np.random.randint(0, 10, size=50)
BMI = np.random.normal(25, 3, size=50)



In [8]:
data = {
    "excersize_hours": exercise_hours,
    "BMI": BMI
}

In [9]:
df = pd.DataFrame(data)
df

excersize_hours        BMI
0                 2  19.704831
1                 2  26.835154
2                 6  31.010078
3                 1  24.911742
4                 3  20.251181
5                 9  24.919867
6                 6  29.715009
7                 1  23.816494
8                 0  29.302686
9                 1  24.723858
10                9  23.439153
11                0  19.906001
12                0  27.304377
13                9  26.180808
14                3  24.048479
15                4  19.891815
16                0  27.215800
17                0  23.476671
18                4  25.683973
19                1  19.778736
20                7  23.396485
21                3  22.186754
22                2  20.541097
23                4  24.911911
24                7  21.288573
25                2  25.227958
26                4  19.164407
27                8  24.295479
28                0  19.556298
29                7  25.178909
30                9  23.807106
31                3  25.803993
32                4  27.672855
33                6  25.076022
34                1  23.205575
35                5  21.308487
36                6  26.340977
37                2  27.251791
38                1  20.876261
39                8  23.357009
40                3  27.646936
41                5  18.548489
42                0  24.746793
43                2  25.028963
44                6  23.420471
45                2  25.492844
46                4  27.983020
47                4  22.645390
48                6  27.728761
49                3  24.034640

In [10]:
## Pearson Corrrelation
df.corr(method="pearson")

excersize_hours       BMI
excersize_hours         1.000000  0.096686
BMI                     0.096686  1.000000

In [12]:
# Calculate Pearson correlation coefficient
pearson_corr, _ = stats.pearsonr(df["excersize_hours"], df["BMI"])
print(f"Pearson correlation coefficient: {pearson_corr}")

Pearson correlation coefficient: 0.09668600178034126


## Interpretation

The `Pearson` correlation coefficient is a measure of the linear relationship between two variables. In this case, the variables are "exercise_hours" and "BMI". The Pearson correlation coefficient between these two variables is 0.096686, which indicates a weak positive correlation between the two variables.

This means that as the number of exercise hours increases, there is a slight tendency for BMI to also increase. However, the correlation is not strong enough to conclude that exercise hours have a significant impact on BMI. Other factors, such as diet and genetics, may also play a role in determining BMI.



### Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between two variables.

In [6]:
import numpy as np
import pandas as pd
from scipy import stats

# Generate random data for exercise hours and BMI
np.random.seed(123) # For reproducibility
exercise_hours = np.random.randint(0, 10, size=50)
BMI = np.random.normal(25, 3, size=50)
data ={
    "exercise_hours": exercise_hours,
    "BMI": BMI
}

df = pd.DataFrame(data)
df

exercise_hours        BMI
0                2  19.704831
1                2  26.835154
2                6  31.010078
3                1  24.911742
4                3  20.251181
5                9  24.919867
6                6  29.715009
7                1  23.816494
8                0  29.302686
9                1  24.723858
10               9  23.439153
11               0  19.906001
12               0  27.304377
13               9  26.180808
14               3  24.048479
15               4  19.891815
16               0  27.215800
17               0  23.476671
18               4  25.683973
19               1  19.778736
20               7  23.396485
21               3  22.186754
22               2  20.541097
23               4  24.911911
24               7  21.288573
25               2  25.227958
26               4  19.164407
27               8  24.295479
28               0  19.556298
29               7  25.178909
30               9  23.807106
31               3  25.803993
32               4  27.672855
33               6  25.076022
34               1  23.205575
35               5  21.308487
36               6  26.340977
37               2  27.251791
38               1  20.876261
39               8  23.357009
40               3  27.646936
41               5  18.548489
42               0  24.746793
43               2  25.028963
44               6  23.420471
45               2  25.492844
46               4  27.983020
47               4  22.645390
48               6  27.728761
49               3  24.034640

In [13]:
## Pearson Corrrelation
df.corr(method="pearson")

excersize_hours       BMI
excersize_hours         1.000000  0.096686
BMI                     0.096686  1.000000

## Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:
## AGE(Years) = [25,42,37,19,31,28]
## Soft drink Preference = ["Coke","Pepsi","Mountain dew","Coke","Pepsi","Coke"]

In [16]:
age = [25,42,37,19,31,28]
soft_drink = ["Coke","Pepsi","Mountain dew","Coke","Pepsi","Coke"]

In [30]:
import numpy as np
from scipy.stats import f_oneway

# Create the data arrays
age = np.array([25, 42, 37, 19, 31, 28])
coke = np.array([1, 0, 0, 1, 0, 1])
pepsi = np.array([0, 1, 0, 0, 1, 0])
mountain_dew = np.array([0, 0, 1, 0, 0, 0])

# Calculate the mean soft drink preference for each age group
mean_coke_young = np.mean(coke[age < 30])
mean_coke_old = np.mean(coke[age >= 30])
mean_pepsi_young = np.mean(pepsi[age < 30])
mean_pepsi_old = np.mean(pepsi[age >= 30])
mean_mountain_dew_young = np.mean(mountain_dew[age < 30])
mean_mountain_dew_old = np.mean(mountain_dew[age >= 30])

# Print the means
print("Mean soft drink preference:")
print("< 30:")
print(f"Coke: {mean_coke_young}")
print(f"Pepsi: {mean_pepsi_young}")
print(f"Mountain Dew: {mean_mountain_dew_young}")
print(">= 30:")
print(f"Coke: {mean_coke_old}")
print(f"Pepsi: {mean_pepsi_old}")
print(f"Mountain Dew: {mean_mountain_dew_old}")
print("")

# Perform the one-way ANOVA
f_value, p_value = f_oneway(coke[age < 30], coke[age >= 30], 
                             pepsi[age < 30], pepsi[age >= 30], 
                             mountain_dew[age < 30], mountain_dew[age >= 30])

# Print the ANOVA results
print("One-way ANOVA:")
print(f"F-value: {f_value}")
print(f"P-value: {p_value}")


Mean soft drink preference:
< 30:
Coke: 1.0
Pepsi: 0.0
Mountain Dew: 0.0
>= 30:
Coke: 0.0
Pepsi: 0.6666666666666666
Mountain Dew: 0.3333333333333333

One-way ANOVA:
F-value: 4.8
P-value: 0.012157439479866439


### Interpretation o f results
- The means show that individuals under 30 years old prefer Coke more than Pepsi or Mountain Dew, while individuals over 30 years old have an equal preference for all three brands.

- The chi-squared test of independence shows that the relationship between age and soft drink preference is not statistically significant (p-value > 0.05). This suggests that there is no association between age and soft drink preference. However, it's worth noting that this is a small sample size, so the results should be interpreted with caution.





## Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [38]:
sales_calls = np.random.randint(50, 100, 30)
sales_made = np.random.randint(10, 50, 30)

data = {
    "sale_calls":sales_calls,
    "sale_made":sales_made
}

df = pd.DataFrame(data)
df

sale_calls  sale_made
0           96         33
1           89         28
2           65         30
3           89         34
4           58         29
5           76         35
6           66         16
7           52         12
8           53         49
9           89         17
10          50         39
11          63         20
12          57         17
13          56         46
14          79         26
15          74         17
16          88         39
17          65         25
18          60         13
19          74         30
20          82         43
21          97         49
22          70         44
23          59         22
24          99         36
25          58         17
26          50         38
27          83         25
28          80         14
29          70         46

In [39]:
# Calculate Pearson correlation coefficient
r, p_value = stats.pearsonr(sales_calls, sales_made)

# Print result
print('Pearson correlation coefficient:', r)

Pearson correlation coefficient: 0.15697197236616392
